docs/reference/analysis/tokenfilters/synonym-tokenfilter.asciidoc

```diff
@@ -4,41 +4,60 @@
 <titleabbrev>Synonym</titleabbrev>
 ++++
 
-The `synonym` token filter allows to easily handle synonyms during the
-analysis process. Synonyms are configured using a configuration file.
-Here is an example:
+The `synonym` token filter allows to easily handle <<search-with-synonyms,synonyms>> during the
+analysis process.
 
-[source,console]
---------------------------------------------------
-PUT /test_index
-{
-  "settings": {
-    "index": {
-      "analysis": {
-        "analyzer": {
-          "synonym": {
-            "tokenizer": "whitespace",
-            "filter": [ "synonym" ]
-          }
-        },
-        "filter": {
-          "synonym": {
-            "type": "synonym",
-            "synonyms_path": "analysis/synonym.txt"
-          }
-        }
-      }
+[discrete]
+[[analysis-synonym-define-synonyms]]
+==== Define synonyms sets
+
+include::synonyms-format.asciidoc[]
+
+[discrete]
+[[analysis-synonym-configure-sets]]
+==== Configure synonyms sets
+
+Synonyms can be configured using the <<synonyms-store-synonyms-api,synonyms API>>, a <<synonyms-store-synonyms-file,synonyms file>>, or directly <<synonyms-store-synonyms-inline,inlined>> in the token filter configuration.
+See <<synonyms-store-synonyms,store your synonyms set>> for more details on each option.
+
+Use `synonyms_set` configuration option to provide a synonym set created via Synonyms Management APIs:
+
+[source,JSON]
+----
+  "filter": {
+    "synonyms_filter": {
+      "type": "synonym",
+      "synonyms_set": "my-synonym-set"
```

Comment:
We also need to add "updateable: true", otherwise a user will get an error.

Reply:
Thanks for the catch! 👍